core/services/job/spawner.go

```diff
@@ -65,6 +65,11 @@ type (
 		// and stopped in reverse order.
 		ServicesForSpec(spec Job) ([]ServiceCtx, error)
 		AfterJobCreated(spec Job)
+		// BeforeJobDeleted will be called from within DELETE db transaction.  Any db
+		// commands issued within BeforeJobDeleted() should be performed first, before any
+		// non-db side effects.  This is required in order to guarantee mutual atomicity between
+		// all tasks intended to happen during job deletion.  For the same reason, the job will
+		// not show up in the db within BeforeJobDeleted(), even though it is still actively running.
```

Comment:
Should it take the open transaction as an arg?

Reply:
Yeah, this is a big oops on my part... I think I had it at one point and then removed it... not sure what I was thinking.  I'm also thinking now that I probably should have had it return an error, which is something I recall going back and forth on.  Between those two, and adding a test to check for this issue, I think this PR Is not as ready as I thought... moving it back into draft for now.